source/common/router/rds_impl.h

```diff
@@ -138,6 +138,7 @@ class RdsRouteConfigSubscription
                       const std::string& system_version_info) override;
   void onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason reason,
                             const EnvoyException* e) override;
+  void onConfigExpired(const std::vector<std::string>&) override {}
```

Comment:
I think in the case of RDS, we should make it an empty route config, EDS, empty CLA, etc.

Reply:
I'm not sure if the additional callback makes this any less likely to be consistently implemented than removals: we already see that lots of xDS clients ignore the remove callback. 

I think the main question is whether we ever want expiration and removal to ever be different, which I've kinda gone back and forth about. I'm starting to think that it might be fine to just handle this all via removal callbacks and have the xDS implementations revert to a null object or an actual removal as appropriate for each resource type?

Reply:
I would vote for using the same callback for both removal and expiration, since I don't think we actually want the ability for a resource type to handle those two events differently.

It looks like you've changed it to do that already, so this looks great!